pkg/build/api/validation/validation.go

```diff
@@ -44,8 +44,16 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) fielderrors.ValidationErr
 		allErrs = append(allErrs, fielderrors.NewFieldInvalid("namespace", config.Namespace, "namespace must be a valid subdomain"))
 	}
 	allErrs = append(allErrs, validation.ValidateLabels(config.Labels, "labels")...)
-	for i := range config.Triggers {
-		allErrs = append(allErrs, validateTrigger(&config.Triggers[i]).PrefixIndex(i).Prefix("triggers")...)
+	// allow only one ImageChangeTrigger for now
+	ictCount := 0
+	for i, trg := range config.Triggers {
+		allErrs = append(allErrs, validateTrigger(&trg).PrefixIndex(i).Prefix("triggers")...)
+		if trg.Type == buildapi.ImageChangeBuildTriggerType {
+			ictCount++
+		}
+	}
+	if ictCount > 1 {
+		allErrs = append(allErrs, fielderrors.NewFieldInvalid("triggers", "", "only one ImageChangeTrigger is allowed"))
```

Comment:
@soltysh why are we passing empty string "" as the `value` ([`ValidationError.BadValue`](https://github.com/GoogleCloudPlatform/kubernetes/blob/907b83cd7d6b400725a1a0d0509d601172920fa4/pkg/util/fielderrors/fielderrors.go#L86)) argument to `fielderrors.NewFieldInvalid`? I've seen this in calls to `fielderrors.NewFieldInvalid` in tests, because we're not checking this value, but I believe we should be creating more meaningful errors? 

And since the error message is going through the API, IMO the long description should talk about "ImageChange trigger", and not "ImageChangeTrigger" -- there's no such name in the API.
```
"triggers": [
  {"type": "ImageChange"}
]
```